docs/adr/0015-admin-api.md

```diff
@@ -14,21 +14,37 @@ Date: 2022-03-04
 * Having static network configuration is probably not sustainable in the long run, even if we don't add any fancy multihead capabilities to the node, as it would make it significantly harder to have automated creation of Heads.
 * There's been an [attempt](https://github.com/input-output-hk/hydra-poc/pull/222) at providing a file-based network configuration but this was deemed unconvincing
 * [Hydra paper (sec. 4, p. 13)](https://eprint.iacr.org/2020/299.pdf) explicitly assumes the existence of a _setup_ phase
+  * This _setup_ is currently left aside, eg. exchanging keys for setting up multisig and identifying peers. The [hydra-node](https://github.com/input-output-hk/hydra-poc/blob/abailly-iohk/admin-api-adr/hydra-node/exe/hydra-node/Main.hs#L41) executable is statically configured and those things are assumed to be known beforehand
 
 ## Decision
 
 * Hydra-node exposes an _Administrative API_ to enable configuration of the Hydra network using "standard" tools
-  * API is exposed as a set of HTTP endpoints on some port, consuming and producing JSON data
+  * API is exposed as a set of HTTP endpoints on some port, consuming and producing JSON data,
   * It is documented as part of the User's Guide for Hydra Head
 * This API provides _commands_ and _queries_ to:
   * Add/remove _peers_ providing their address and keys,
   * List currently known peers and their connectivity status,
   * Start/stop/reset the Hydra network
+* This API is implemented by a _new component_ accessible through a network port separate from current _Client API_, that _configures_ the `Network` component
+
+The following picture sketches the proposed architectural change:
+
+![Architecture change](./0015-architecture-change.jpg)
+
+## Q&A
+
+* *Why a REST interface?*
+  * This API is an interface over a specific _resource_ controlled by the Hydra node, namely its knowledge of other peers with which new _Head_s can be opened. As such a proper REST interface (_not_ RPC-in-disguise) seems to make sense here, rather than stream/event-based [duplex communication channels](./0003-asynchronous-duplex-api.md)
+  * We can easily extend such an API with websockets to provide notifications (eg. peers connectivity, setup events...)
+* *Why a separate component?*
+  * We could imagine extending the existing [APIServer](../../hydra-node/src/Hydra/API/Server.hs) interface with new messages related to this network configuration, however this seems to conflate different responsibilities in a single place: Configuring and managing the Hydra node itself, and configuring, managing, and interacting with the Head itself
```

Comment:
I wouldn't say it conflates these responsibilities, but the API component would merely *provide access* to the functionality of managing the hydra-node. The same way as the current functionality of the hydra-node (manage a single head and use it when open) is also not provided by the API component right now.

Reply:
We are talking about managing 2 different entities, with clear separation of capabilities and state: The Node and its network part on one side, the Head which the Node is hosting on the other side (there can be multiple Heads in a Node, possibly with different parties and peers, if not concurrently atm at least sequentially). 
Hence it seems that having a single component (the current APIServer) providing access to these 2 capabilities is a conflation of 2 different concerns.